.github/workflows/process_new_pull_request.yml

```diff
@@ -0,0 +1,95 @@
+name: Process New Pull Request
+
+on:
+  workflow_run:
```

Comment:
I think the idea is that this only triggers when a PR is opened, and not otherwise.

When this workflow triggers, it should run both the oppiabot job and the assign_reviewers job.

Reply:
When a PR is opened, the Oppiabot job is triggered first to verify the CLA. Once the Oppiabot job completes, this workflow runs on the same PR (this workflw requires the oppiabot job to complete successfully)

If we want both the Oppiabot job and the Assign Reviewers job to run together when a PR is opened, we would need to include the Oppiabot job within this workflow file itself.

Should I make this change to combine both jobs in a single workflow file, or is the current setup—where they are in separate workflows—works? Please let me know.

and 

PS-> When a PR is opened by a new contributor who hasn’t signed the CLA, the Oppiabot job fails, which in turn prevents this workflow from running since it as of now depends on oppiabot job to complete successfully.

To address this, should I make this workflow independent of the Oppiabot job? This way, even if the Oppiabot job fails, the action can guide new contributors by adding a comment using the `PTAL` syntax to request a reviewer. 

Let me know if this approach works.

Reply:
Hm, good points. I think what we really should be doing is to split Oppiabot into a collection of reusable actions that can then be imported into our workflows. The problem now is that all of them are in a single action and I think that makes things harder.

So yeah, let's just keep this action separate from oppiabot and drop the on:workflow_run part. There's no point running this workflow anyway if the oppiabot job fails (in such cases we should actually auto-close this PR).

Reply:
yes this works.
so i am droping the on:workflow_run part and rest keeping same , do i need to make some more additions.